internal/deepcopy/deepcopy.go

```diff
@@ -0,0 +1,23 @@
+package deepcopy
+
+// DeepCopy performs a recursive deep copy for nested slices/maps and
+// returns the copied object. Supports []interface{}
+// and map[string]interface{} only
+func DeepCopy(val interface{}) interface{} {
```

Comment:
nit: I'd add some test coverage here now that this is exposed for re-use within the rest of OPA.